Pages and UI update

jasonleyser opened the pull request:
- Added modal page states for the home, account and shortcut pages.
- Bug fixes
- Added slate logo back
- Added Inter font
- CSS variables
- Progress bar for account storage

**Pages**

Main modal
<img width="667" alt="Screen Shot 2021-09-13 at 3 50 06 PM" src="https://user-images.githubusercontent.com/60402678/133161571-9323ded9-c286-48d6-9f91-2cb37f49c3b8.png">

Account
<img width="666" alt="Screen Shot 2021-09-13 at 3 50 20 PM" src="https://user-images.githubusercontent.com/60402678/133161595-58f8390d-a21a-4715-9978-f040bd12e702.png">

Shortcuts
<img width="668" alt="Screen Shot 2021-09-13 at 3 50 12 PM" src="https://user-images.githubusercontent.com/60402678/133161615-80a80bd6-4196-468d-bbc4-44518c25c131.png">

jasonleyser committed:
Pages and UI update

- Added the modal page states for the home, account and shortcut pages.
- Bug fixes
- Added slate logo back
- Added Inter font
- CSS variables
- Progress bar for account storage

martinalong commented:
![image](https://user-images.githubusercontent.com/33686587/133345288-9958c627-aa1c-4b31-8d77-dd4493e3706c.png)
Looks like it's still misaligned on some websites such as https://medium.com/yom-ai/rest-api-with-node-js-and-elasticsearch-1368cf9df02a
Adding display: flex to .modalButtonSVG seemed to solve it for me

martinalong commented:
We shouldn't be committing the build folder to git because it's a file that's compiled from the other files. Instead, add `build/` to the .gitignore file and delete the build files that you've already pushed to git, then push again
https://community.frontity.org/t/git-commit-for-build-directory/466/4

martinalong commented:
Can you run npm install prettier and then go into every file and hit ctrl + s? This will auto format it with prettier, and help us avoid awkward formatting like this: 
![image](https://user-images.githubusercontent.com/33686587/133350498-942d165a-6345-4594-8877-31e7b164f066.png)

martinalong commented:
There's a few organizational standards that we use in the main app that would be good to use in the extension as well. Right now, they may not be significant since the extension is relatively small, but as it grows it'll help keep everything neat.

- Use @emotion/css 
    - first, `npm install @emotion/css`
    - then you can use it like we do in the regular app with import { css } from "@emotion/css". The purpose of this is to be able to use in page styles instead of having one giant css file that's hard to parse

Code reuse:
The purpose of putting repeated code or values into a shared file and then importing it is so that we can save on code, and so that when we need to make updates to values or code, we only have to do so in one place rather than having to hunt for every instance of it. Imagine trying to control + F all the places where we used a certain color in order to change it, versus changing one line in a contants file.
- put the svg.js file (along with any other shared "utilities") in a folder at the root level called `common` (aka common/svg.js, common/constants.js)
- you will probably also need a constants.js file in this common folder. It's up to you what you put in there, but the trick is to always be looking out for reused values that could belong in it. at the very least, put all the colors that you are using in there, like what we do in the main app
- you may also need a strings.js file in the common folder which is for commonly reused string manipulations (like getting a url from a cid), or a validitions.js which is for commonly reused checks (like checking if a username is valid). For examples, see the main app code

martinalong commented on src/Components/Button.js:
```diff
@@ -1,23 +1,34 @@
 	let svg;
-	if(props.icon == 'tag') {
-		svg = <Icons.Tag />
+	if(props.icon == 'command') {
```
Instead of this complicated else if structure, you can just create an object matching names to svg's, and it'll also be faster (since looking up something in a hashtable--which is essentially what an object is--is faster than stepping through a bunch of if statements sequentially)

```
const icons = {
  bookmark: <Icons.Bookmark />,
  plus: <Icons.Plus />,
  ...
}

svg = icons[props.icon];
```

martinalong commented on src/Components/Button.js:
```diff
@@ -1,23 +1,34 @@
 import React, { useState } from 'react';
```
I find it neatest to organize imports by "style" and put a space in between the different styles, such as:
```
import React, { useState } from "react";

import * as Icons from "../Components/Icons";
import * as Strings from "/common/strings";

import classes from "../App.module.css";
```

martinalong commented on src/Components/Button.js:
```diff
@@ -1,23 +1,34 @@
 import React, { useState } from 'react';
 import * as Icons from '../Components/Icons';
 import classes from '../App.module.css';
+require('typeface-inter');
```
Don't mix require and import syntax. They are equivalent, but import syntax is the new es6 standard, so this should be written as `import Inter from "typeface-inter"`

martinalong replied:
However, I think here you should stick to using font faces instead of using a package specifically just to load this font.

You would do that by adding it as a global style with:
```
  @font-face {
    font-family: 'inter-regular';
    src: url('https://slate.textile.io/ipfs/bafkreic3hkcuwvrmf6trweqcjp62valsfjh3zvwacikoreynakgw67wrvy');
  }
```

Fonts can either be loaded from a local file (like the one you have in static) or from a url like the one above. I'd recommend loading from a url like we do in the main app bc then it can load from that url simultaneously while the rest of the app loads from a different server, rather than competing for the same "loading bandwidth" from a single source. This would mean you can delete the Inter-Regular.ttf font from the public folder

jasonleyser replied:
for security reasons chrome doesn't allow linking to am external url for anything like css, scripts, fonts and all that.

martinalong commented:
Can you delete the unused logo.svg, tag.svg, and other svg files now that you're just exporting from the svg.js folder instead? Same goes for the unused logo images like icon-128.png:
![image](https://user-images.githubusercontent.com/33686587/133352619-40795581-f331-401e-8d7c-81cbb3cee603.png)

martinalong commented:
What are public/settings.html and public/settings.js for? Curious why they're in the public folder

martinalong commented on src/Components/Button.js:
```diff
@@ -32,6 +43,12 @@ const Button = (props) => {
+					{props.command &&
+						<span className={classes.modalCommandIcon}>⌥</span> 
+					}
+					<span className={classes.modalKeyIcon}>{props.shortcut}</span>
```
For the styling for this, instead of applying the background and border radius directly to the text, i'd recommend putting the text in a div and then playing around with height/width and padding to make it look more like the designs. The background is currently much more narrow than in the designs.

Implementation: 
![image](https://user-images.githubusercontent.com/33686587/133519028-8af4f8ab-341f-4be9-ad80-67721cc5deff.png)

The design:
![image](https://user-images.githubusercontent.com/33686587/133519120-dc7b001f-f0b7-4517-bcf0-3ca9033673ea.png)

martinalong commented:
The extension has a weird bug on this page: https://developer.chrome.com/docs/extensions/reference/cookies/
![image](https://user-images.githubusercontent.com/33686587/133519599-78e7670c-9f63-43be-b0c1-e93056cf846a.png)

martinalong commented:
![image](https://user-images.githubusercontent.com/33686587/133519694-3c09776b-ea2f-47db-afab-a633507f0741.png)
The slate symbol for when there isn't an image seems to have a thin line under it

martinalong commented:
And on dropbox, the modal goes to one side:
![image](https://user-images.githubusercontent.com/33686587/133519841-89dd2ae5-960b-454c-8040-93d9c7765e34.png)

martinalong commented:
![image](https://user-images.githubusercontent.com/33686587/133519901-e25e9b1d-af8f-44d0-b3bc-8204532b4569.png)
font sizing on the description is too large on substack

martinalong commented:
> The extension has a weird bug on this page: https://developer.chrome.com/docs/extensions/reference/cookies/
> ![image](https://user-images.githubusercontent.com/33686587/133519599-78e7670c-9f63-43be-b0c1-e93056cf846a.png)

![image](https://user-images.githubusercontent.com/33686587/133519960-e509a0f6-15d2-4657-9cfb-b09f33a3c00e.png)
More image sizing issues on knownorigin.io

martinalong commented:
On some dropbox links, it also seems to scroll with the page instead of being static: https://www.dropbox.com/sh/4mm6if0if9j01g4/AADMniHJGxsHmdRL_Smd1PFva?dl=0

jasonleyser committed:
UI fixes, cookies and code cleanup

UI fixes, cookies and code cleanup

jasonleyser commented:
@martinalong 

The recent update to this PR includes:
- code cleanup (deleted old files in public and unused components)
- added a common folder in /src including: constants, strings, styles and svgs 
- added shadow dom and new css structure (emotion doesn't seem to be working so right now its in a constructable stylesheet, but im still working around this)
- ran prettier on all files
- added cookies and fetching api key with login session


https://github.com/slate-engineering/slate-web-extension/pull/4/commits/18351bf59565a43fd3af0b1bc84ce073a1a7d769

martinalong commented:
![image](https://user-images.githubusercontent.com/33686587/134730030-d4812c33-6676-4f50-87ea-c4e9ec790469.png)
The add to library button still seems misaligned. The text is too low compared to the icon

martinalong commented:
![image](https://user-images.githubusercontent.com/33686587/134730492-ffc33b0b-ec3b-4d9b-9a40-357a37fe9fe2.png)
The account icon also looks thinner than the rest and slightly to the right, likely because strokeWidth is not specified for it:
![image](https://user-images.githubusercontent.com/33686587/134731063-f1435837-472b-4b16-9847-e8929e934340.png)

Also, no need to keep all the SVGs that the slate application has in the SVG.js file. Only the ones that the extension uses

martinalong commented:
![image](https://user-images.githubusercontent.com/33686587/134730542-ba7e06ce-7f1c-47e4-b2ee-d3a4205014f2.png)
Sometimes the image shows as a broken image rather than the default slate logo when no image is found, like for this page: https://github.com/brianc/node-postgres/issues/442

martinalong commented:
Also, a lot of pages which I have never saved show up with a green checkmark indicating that they have been saved. Can you track down the cause of this? And if it's the endpoint returning incorrect values, let me know and I'll fix it

I've definitely never saved this link to slate, but it shows a green check:  
![image](https://user-images.githubusercontent.com/33686587/134730704-24d144fd-71a4-459f-abd9-1fbd9a61a3f7.png)

jasonleyser committed:
Bug fixes, component updates

- Fixed bug where links are detected as uploaded but they are not.

- Updated toast loader

- added login/logout state

jasonleyser committed:
Added login check for shortcuts

Added user login checks when the extension is opened with a shortcut

martinalong commented:
![image](https://user-images.githubusercontent.com/33686587/135195622-a2235f6d-d8af-4453-96e1-ccdd102d2166.png)

I'm getting this error when I tried to build it. Just a reminder to delete that unused import:
![image](https://user-images.githubusercontent.com/33686587/135195673-4a107b09-f85c-4fae-b83b-cd1f0c9fd30c.png)

martinalong commented:
![image](https://user-images.githubusercontent.com/33686587/135197342-5442dfb4-7a38-412d-8168-e5f2d5bc23fe.png)
On some websites the url is a really large font for some reason. This is the site: https://askanyc.com/about/

But then on others it's smaller https://www.exploretock.com/perse/:
![image](https://user-images.githubusercontent.com/33686587/135198200-9ea114c7-adf9-4ca9-9a3f-0393e6982475.png)

martinalong commented:
https://user-images.githubusercontent.com/33686587/135197456-5783663a-af5c-494c-9874-a29f89b2c749.mov

The loader spinner moves around as it spins. Take a look at how we do it in the main application. It may be b/c the SVG is not centered in the box that is rotating, or b/c SVGs have a bit of space underneath them if you don't do display: "block" (this happens b/c they are inline elements by default, and inline elements leave a bit of space underneath them for the parts of letters that go below the "baseline")

martinalong commented:
The loader spinner moves around as it spins. Take a look at how we do it in the main application. It may be b/c the SVG is not centered in the box that is rotating, or b/c SVGs have a bit of space underneath them if you don't do display: "block" (this happens b/c they are inline elements by default, and inline elements leave a bit of space underneath them for the parts of letters that go below the "baseline"

I slacked you a video (didn't show up well on github)

martinalong commented:
Saved links now don't seem to show up as saved (opposite problem of last time, where everything showed up as a green check mark)

![image](https://user-images.githubusercontent.com/33686587/135198440-9afc8099-c777-4355-8221-5198d6e9f903.png)

jasonleyser committed:
UI and api bug fixed

fixes:
- loading spinner
- link check works again
- open link if its already been saved
- metadata url font size
- removed unused component

martinalong commented:
![image](https://user-images.githubusercontent.com/33686587/135735222-8a69e375-46c8-4926-b1ef-1d138a6e0440.png)
There's some extra space below the SVG here. I think this may be one where putting "display: block" on the svg will solve it

martinalong commented:
Can we make the shortcuts on the shortcuts page the same color as the ones on the main page? Aka grey text?
![image](https://user-images.githubusercontent.com/33686587/135735242-aae86a38-f49f-452e-bb8a-77c387669c5c.png)

martinalong commented:
![image](https://user-images.githubusercontent.com/33686587/135735279-0f3edc43-4ac3-44f8-b222-bc36bee4b04c.png)
This is just dummy info right?

martinalong commented:
![image](https://user-images.githubusercontent.com/33686587/135735344-765a1a52-153b-4d50-a6fc-b0a1c6236532.png)
Loader and the word "saving" are misaligned

martinalong commented:
I noticed the option + B shortcut (bookmark the page) only works if you already have the extension open. Just typing it normally doesn't seem to trigger a save. Can we make it so it works even without the extension open? Or is there some restriction on it?

martinalong commented:
![image](https://user-images.githubusercontent.com/33686587/135735393-572dae11-c6b0-4cca-8ae5-b0528be469f7.png)
Can we line these two pieces of text up? "welcome to slate for chrome" should have the same left margin as "slate web extension"

![image](https://user-images.githubusercontent.com/33686587/135735411-d6751209-82a3-4b0b-b3f3-604fcacbb2e4.png)
And sign in should be bolded like our buttons in the main application

martinalong commented:
![image](https://user-images.githubusercontent.com/33686587/135735450-ac1b7357-919d-4c64-9445-23edca173a32.png)
When something already exists on slate, hitting "enter" doesn't take you to the app like it's supposed to. Instead, it just tries to save it, then says it already exists:
![image](https://user-images.githubusercontent.com/33686587/135735464-4af2d38a-c9a7-49e2-9e14-72adb25ea344.png)

jasonleyser committed:
Bug fixes and image check

- Added a way to check to see if an image has loaded, some sites dont allow cross domain loading so a fallback div is now used
- bug fixes and some css changes
- you can now use enter to view a link that has been uploaded

jasonleyser committed:
Auto refresh on login

The extension now auto updates after you login to the web app. If the session happens from the extension, the browser will go back to the previous tab you were on

martinalong commented:
![image](https://user-images.githubusercontent.com/33686587/136867347-a1a3db76-2e29-47cd-83f2-1c4ec2b098b8.png)
The reason your SVGs are unevenly spaced is because they are unevenly sized. This one is wider than the SVG above it (which you seem to be trying to fix with a margin left -3) but that doesn't address the extra width on the right side of it which is now pushing the word "account" to the right.

For all SVGs, the way to standardize them is by putting a set width and height container around it and centering it within that container. Then you don't have to worry about these variations or do hacky things like negative margins. 

If you ever find yourself measuring the positioning of a certain thing and hard coding to adjust it, there's probably a better option. You're always welcome to ask whether there's a "best practice" way to do something, it's one of the best ways to learn--no need to wait till there's a bug to ask questions!

```
//we actually have this style saved in our styles file as STYLES_ICON_CONTAINER
.svgcontainer {
  width: 24px;
  height: 24px;
  display: flex;
  align-items: center;
  justify-content: center;
}

<div classname="svgcontainer">
  <SVG />
</div>
```

martinalong commented:
![image](https://user-images.githubusercontent.com/33686587/136867833-ca348b00-520e-4b98-8688-f328dedceb07.png)

For this component, one thing you could do is do:
```
display: grid;
grid-template-columns: 50px 1fr auto; //replace the 50px with whatever it actually is
```
This will give the left side icon a fixed width of 50px, the right side will take up however much space it needs, and the middle section will take up whatever is left. This will help you keep each row standard, rather than worrying about aligning each row individually and hoping they match

martinalong commented:
The option + B bookmark this page still doesn't seem to work

martinalong commented:
![image](https://user-images.githubusercontent.com/33686587/136868028-333ed84b-6bc9-4a9d-abcc-12ea3d57d204.png)
You can remove the uploads section since we've decided to no longer have a section to view your upload history

martinalong commented:
![image](https://user-images.githubusercontent.com/33686587/136868175-7435b686-5c63-4f54-a28a-68d8e55388c2.png)
Let's also remove this shortcut. Seems unnecessary

jasonleyser committed:
Multiple loader styles, css updates

Updates:
- added a small loader for quick save
- large loader for jumper open
- css updates

jasonleyser committed:
public folder

pushing the public folder with manifest, background and content file updates

martinalong commented:
![image](https://user-images.githubusercontent.com/33686587/139721567-f9a79dcf-49c7-4f56-ae1b-8f60713128fe.png)
On the right side there's the grey background for a shortcut, but no shortcut. let's remove the lingering grey background

jasonleyser committed:
Removed grey background for empty shortcut

Removed the grey background if a button has an empty shortcut
